content/sctk-simple-image-viewer.markdown

```diff
@@ -0,0 +1,657 @@
+
+fn main() {
+    // ...
+
+    let (display, mut event_queue) = Display::connect_to_env()
+                                        .expect("Failed to connect to the Wayland server.");
+
+    // ...
+}
+```
+
+This `connect_to_env()` function tells Wayland-client to try to connect to a Wayland
+server according to the configuration of the environment variables. This is what most
+Wayland clients are expected to do.
+
+Now, from this client, we are supposed to retrieve the `wl_registry` and process its
+events to retrieve the global list and bind the globals we need. For this, SCTK provides
+us with a first utility that does that for us: `sctk::Environment`. It takes a registry
+as an argument, wraps it, processes its events and automatically binding the most classic
+globals. This will be sufficient for our example.
+
+```rust
+use sctk::Environment;
+// our display object is actually of type `Proxy<WlDisplay>`, so we need to import
+// this extension trait to access methods representing its requests
+use sctk::reexports::client::protocol::wl_display::RequestsTrait as DisplayRequests;
+
+fn main() {
+    // ..
+
+    // All request method of Wayland objects return a `Result`, as Wayland-client
+    // returns an error if you try to send a message on an object that has already
+    // been destroyed by a previous message. We know our display has not been
+    // destroyed, so we unwrap().
+    let registry = display.get_registry().unwrap();
+
+    // The Environment takes a registry and a mutable borrow of the Wayland event
+    // queue. It uses the event queue internally to exchange messages with the server
+    // to process the registry event.
+    // Like all manipulation of the event loop, it can fail (if the connection is
+    // unexpectedly lost), and thus returns a Result. This failing would prevent
+    // us to continue though, so we unrap().
+    let env = Environment::from_registry(registry, &mut event_queue).unwrap();
+
+    // ..
+}
+```
+
+Once `Environment::from_registry(..)` returns, we get an already-initialized `Environment`
+with various Wayland globals already bound, available as public fields. First of all,
+we will use the `wl_compositor` global to create a `wl_surface`. The surfaces are the most
+fundamental building block of a Wayland app. It describe some rectangle of pixels to which
+you can attach a `wl_buffer` defining its contents. A Wayland app is built on surfaces:
+
+- Most often, they can be assigned a role. The most classic one would be "toplevel surface",
+  meaning this surface is actually a window in the traditional sense. But surfaces can also
+  be used to change the image of the mouse pointer for example.
+- They can be defined relative to a parent surface (they are then a "subsurface"), this can
+  typically be usefull if different parts of your UI are drawn by different means. For example
+  a video player could use GPU-accelerated graphism to decode and display the contents of the
```

Comment:
I don't know the word "graphism" and can't find a translation for it. What did you mean here?